File: app/src/main/java/com/rainstar/demo/spi/java/StrategyManager.java
package com.rainstar.demo.spi.java;

import com.rainstar.demo.spi.java.SpiDemoInterface;

/**
 * Created by rainstar on 2023/7/11.
 */
public class StrategyManager {
    private SpiDemoInterface mSpiDemoInterface;

    public StrategyManager(SpiDemoInterface spiDemoInterface) {
        this.mSpiDemoInterface = spiDemoInterface;
    }

    public void displayName(String mode) {
        mSpiDemoInterface.displayName(mode);
    }

    public String getDesc(String mode) {
        return mSpiDemoInterface.getDesc(mode);
    }
}

File: app/src/main/java/com/rainstar/demo/spi/kt/StrategyManager.kt
package com.rainstar.demo.spi.kt

/**
 * Created by rainstar on 2023/7/11.
 */
class StrategyManager(spiDemoInterface: SpiDemoInterface) {
    private var mSpiDemoInterface: SpiDemoInterface

    init {
        mSpiDemoInterface = spiDemoInterface
    }

    fun displayName(mode: String) {
        mSpiDemoInterface.displayName(mode)
    }

    fun getDesc(mode: String): String {
        return mSpiDemoInterface.getDesc(mode)
    }
}